rep-2004.rst

```diff
@@ -0,0 +1,527 @@
+
+* Documentation:
+
+  * Must have a declared license or set of licenses
+  * Must have a copyright statement in each source file
+  * May have a "quality declaration" document, which declares the quality level and justifies how the package meets each of the requirements
+
+    * Must have a section in the repository's ``README`` which contains the "quality declaration" or links to it
+    * May register with a centralized list of 'Level 3' packages, if one exists, to allow for peer review of the claim
+
+  * [ROS Core]:
+
+    * Must have automated checks for copyright statements and licenses
+    * Must use the Apache 2.0 license, unless the package has an existing permissive license (e.g. rviz uses three-clause BSD)
+
+* Testing:
+
+  * No explicit testing requirements, though covering some if not all of the features with tests is recommended
+
+* Dependencies:
+
+  * May have direct runtime "ROS" dependencies which are not 'Level 3' dependencies, but they should be documented
+
+* Platform Support:
+
+  * Must support all tier 1 platforms for ROS 2, as defined in `REP-2000 <https://www.ros.org/reps/rep-2000.html#support-tiers>`_
+
+If the above points are satisfied then a package can be considered 'Level 3'.
+Refer to the detailed description of the requirements in the Quality Level 1 section above for more information.
+
+Quality Level 4
+^^^^^^^^^^^^^^^
+
+These are demos, tutorials, or experiments.
+They don't have strict requirements, but are not excluded from having good documentation or tests.
+For example, this might be a tutorial package which is not intended for reuse but has excellent documentation because it serves primarily as an example to others.
+
+Package Requirements
+~~~~~~~~~~~~~~~~~~~~
+
+*Note: bullets below that start with [ROS Core], will be the prescription for what we do in the core packages in order to meet the associated requirements*
+
+Requirements to be considered a 'Level 4' package:
+
+* Version Policy:
+
+  * No requirements, but having a policy is still recommended (e.g. ``semver``), even if the version is not yet stable (e.g. >= 1.0.0 for ``semver``)
+
+* Change Control Process:
+
+  * No explicit change control process required, but still recommended
+
+* Documentation:
+
+  * Must have a declared license or set of licenses
+  * Must have a copyright statement in each source file
+  * [ROS Core]:
+
+    * Must have automated checks for copyright statements and licenses
+    * Must use the Apache 2.0 license, unless the package has an existing permissive license (e.g. rviz uses three-clause BSD)
+
+* Testing:
+
+  * No explicit testing requirements, though covering some if not all of the features with tests is recommended
+
+* Dependencies:
+
+  * No restrictions
+
+* Platform Support:
+
+  * May support all tier 1 platforms for ROS 2, as defined in `REP-2000 <https://www.ros.org/reps/rep-2000.html#support-tiers>`_
+
+Any package that does not claim to be 'Level 3' or higher is automatically 'Level 4'.
```

Comment:
This may again be my understanding of English, but: does "higher" mean higher quality, or at a level with a higher (larger?) number?

```suggestion
Any package which does not claim to be 'Level 3' or higher is automatically 'Level 4'.
```

Reply:
I meant higher standard, which in this case means a lower number. I can rephrase it to be clearer.

I'd also be open to renaming/rewording this jargon to be less confusing, maybe we could just letters, or something else.

Reply:
```suggestion
Any package that does not claim to be 'Level 3' or higher quality is automatically 'Level 4'.
```
"that" is correct here, not "which"